File: vision_visp/visp_auto_tracker/src/point_from_realsense.cpp
#include "ros/ros.h"
#include "geometry_msgs/PoseStamped.h"

#include <iostream>
 
using namespace std;

// 接收到订阅的消息后，会进入消息回调函数
void chatterCallback(const geometry_msgs::PoseStamped::ConstPtr& msg)
{
    double x=0, y=0, z=0, qx = 0, qy = 0, qz = 0, qw = 0;
    x = msg-> pose.position.x;
    y = msg-> pose.position.y;
    z = msg-> pose.position.z;
    qx = msg-> pose.orientation.x;
    qy = msg-> pose.orientation.y;
    qz = msg-> pose.orientation.z;
    qw = msg-> pose.orientation.w;

    cout << "---" << endl;
    cout << "pose: " << endl;
    cout << "  position: " << endl;
    cout << "\tx: " << x << endl;
    cout << "\ty: " << y << endl;
    cout << "\tz: " << z << endl;
    cout << "  orientation: " << endl;
    cout << "\tx: " << qx << endl;
    cout << "\ty: " << qy << endl;
    cout << "\tz: " << qz << endl;
    cout << "\tw: " << qw << endl;
    cout << "---" << endl;

}

int main(int argc, char **argv)
{
  // 初始化ROS节点
  ros::init(argc, argv, "listener");

  // 创建节点句柄
  ros::NodeHandle n;

  // 创建一个Subscriber，订阅名为chatter的topic，注册回调函数chatterCallback
  ros::Subscriber sub = n.subscribe("/visp_auto_tracker/object_position", 1000, chatterCallback);

  // 循环等待回调函数
  ros::spin();

  return 0;
}

File: vision_visp/visp_auto_tracker/scripts/point_from_realsense.py
#!/usr/bin/env python 
# -*- coding: utf-8 -*- 

import rospy
from geometry_msgs.msg import PoseStamped
from std_msgs.msg import String
from std_msgs.msg import Float64  
import numpy

def callback(pose): 
    global x, y, z, qx, qy, qz, qw
    x = pose.pose.position.x
    y = pose.pose.position.y
    z = pose.pose.position.z 
    qx = pose.pose.orientation.x
    qy = pose.pose.orientation.y
    qz = pose.pose.orientation.z
    qw = pose.pose.orientation.w
    print '---\n pose: \n  position: \n\t x: %.12f \n\t y: %.12f\n\t z: %.12f\n  orientation: \n\t x: %.12f\n\t y: %.12f\n\t z: %.12f\n\t w: %.12f \n---' %(x, y, z, qx, qy, qz, qw)

def listener(): 
    rospy.init_node('listener', anonymous=True) #初始化节点，节点名称
    pose=PoseStamped()     
    rospy.Subscriber('/visp_auto_tracker/object_position', PoseStamped, callback,queue_size = 1)
    rospy.spin()

if __name__ == "__main__":
    listener()

